File: frontend/src/types/risk-profile.ts
import type { Finding } from "./finding";

export interface RiskProfile {
  person_id: string;
  name: string;
  email: string;
  score: number;
  risk_level: string;
  findings: string[];
}

export interface RiskProfileDetailResponse {
  profile: RiskProfile;
  findings: Finding[];
  ai_report: {
    person_id: string;
    risk_score: number;
    risk_level: string;
    summary: string;
    security_impact: string;
    recommended_actions: string[];
  } | null;
}

File: models/risk_profile.py
"""Risk profile model for prioritized identity scoring."""

from __future__ import annotations

from dataclasses import dataclass, field
from typing import List


@dataclass
class RiskProfile:
    person_id: str
    name: str
    email: str
    score: int
    risk_level: str
    findings: List[str] = field(default_factory=list)

    def to_dict(self) -> dict:
        return {
            "person_id": self.person_id,
            "name": self.name,
            "email": self.email,
            "score": self.score,
            "risk_level": self.risk_level,
            "findings": list(self.findings),
        }
